File: euclidean-algorithm/euclidean_algorithm.py
def gcd(a, b):
    """Greatest Common Divisor using Euclidean Algorithm"""
    while b:
        a, b = b, a % b
    return a

def gcd_recursive(a, b):
    """Recursive version"""
    if b == 0:
        return a
    return gcd_recursive(b, a % b)

def lcm(a, b):
    """Least Common Multiple"""
    return abs(a * b) // gcd(a, b)

def extended_gcd(a, b):
    """Extended Euclidean Algorithm"""
    if a == 0:
        return b, 0, 1
    
    gcd_val, x1, y1 = extended_gcd(b % a, a)
    x = y1 - (b // a) * x1
    y = x1
    
    return gcd_val, x, y

# Test
print(f"GCD(48, 18) = {gcd(48, 18)}")  # 6
print(f"LCM(48, 18) = {lcm(48, 18)}")  # 144

File: euclidean-algorithm/euclidean_algorithm.rb
def gcd(a, b)
  while b != 0
    a, b = b, a % b
  end
  a
end

def lcm(a, b)
  (a * b) / gcd(a, b)
end

def extended_gcd(a, b)
  return [a, 1, 0] if b == 0
  
  g, x1, y1 = extended_gcd(b, a % b)
  x = y1
  y = x1 - (a / b) * y1
  
  [g, x, y]
end

puts "GCD: #{gcd(48, 18)}"
puts "LCM: #{lcm(48, 18)}"
puts "Extended GCD: #{extended_gcd(48, 18)}" 